### PR TITLE
Increase version for jakarta-activation/mail

### PR DESCRIPTION
**Pull request by matthiaso**
Including Eclipse Angus implementation (as jakarta dependencies now only
 include the specification).

391137

### CODE REVIEW

File: org.eclipse.scout.rt/pom.xml [hunk 1]
```diff
@@ -158,9 +158,14 @@
...
-        <groupId>com.sun.mail</groupId>
-        <artifactId>jakarta.mail</artifactId>
-        <version>2.0.1</version>
+        <groupId>jakarta.mail</groupId>
+        <artifactId>jakarta.mail-api</artifactId>
+        <version>2.1.3</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.angus</groupId>
+        <artifactId>angus-mail</artifactId>
```
Comment (matthiaso) [inline org.eclipse.scout.rt/pom.xml:168]: Does it include the API? Do we need an additional API dependency (before this one)?

File: org.eclipse.scout.rt.rest/pom.xml [hunk 2]
```diff
@@ -35,16 +35,20 @@
       <groupId>jakarta.xml.bind</groupId>
       <artifactId>jakarta.xml.bind-api</artifactId>
       <exclusions>
-        <!-- Remove the activation-api, Scout includes the implementation anyway, and it includes the api classes as well. -->
         <exclusion>
-          <groupId>jakarta.activation</groupId>
-          <artifactId>jakarta.activation-api</artifactId>
+          <!-- jakarta.activation:jakarta.activation-api should be included instead -->
+          <groupId>com.sun.activation</groupId>
+          <artifactId>jakarta.activation</artifactId>
```
Comment (matthiaso) [inline org.eclipse.scout.rt.rest/pom.xml:40]: Previous Jakarta EE 10 excluded? Now Jakarta EE 8 excluded, check again.

**Comment by matthiaso** [comment]
Verify API old vs. new

File: org.eclipse.scout.rt.rest/pom.xml (hunk 2 above)
Comment (matthiaso) [reply org.eclipse.scout.rt.rest/pom.xml:40]: See https://repo1.maven.org/maven2/jakarta/xml/bind/jakarta.xml.bind-api/3.0.1/jakarta.xml.bind-api-3.0.1.pom vs. https://repo1.maven.org/maven2/jakarta/xml/bind/jakarta.xml.bind-api/2.3.3/jakarta.xml.bind-api-2.3.3.pom (previous exclusion was not necessary anymore for version 3.0.1) - however for current version 4.0.0 upwards, the newer jakarta.activation-api is included again. :-)

**Review by LukasHuser — APPROVED** [review]

**Pull request merged**